components/shared/interaction/delete.tsx

```diff
@@ -0,0 +1,29 @@
+'use client';
+
+import { DeletePost } from '@/app/api/actions/post.actions';
+import { Post } from '@/utils/models';
+import { CancelIcon, Toggle } from 'clada-storybook';
+
+const copyLinkLabels = {
+  default: 'Copy Link',
+  active: 'Link copied',
+};
+
+export const DeleteUserPost = ({ post }: { post: Post }) => {
+  const deletePost = () => {
+    DeletePost(post.id, post.creator.id);
```

Comment:
Question: any reason this function is not async?

Reply:
not any good ones. I changed it. Good spot!